### PR TITLE
Added ES6 feature support to NodeJS Native codegen

### PR DESCRIPTION
**Pull request by sastava007**
**What this PR does**
It allows the user to generate the code snippet for NodeJS Native in ES6 format.
This is in response to issue #115  requested by one of the users.

**Changes you made**

- Registered a new parameter ES6_enabled in **`options()`** which allows the user to toggle between whether codegen generates snippet with ES6 features or not.

- By default ES6_enabled is set to `false`

- Also added the newly created option parameter in **`structure.test.js`** file to register it as a valid option id.

- Added unit tests to verify the generated snippets in **snippet.test.js**

- Added a new mocha test suite to verify that the generated snippet is runnable in **newman.test.js.**

- Updated the **readme.md** file with new option parameter.

**Screenshot/ CodeSnippet**
Not required

**Test Configuration**:

- Added a unit test to check the snippet generated

- Added a new block of mocha test inside the main suite to verify that the snippet produced is runnable.

**Follow up**
Not required

*Thanks for contributing!* ❤️

### CODE REVIEW

**Comment by sastava007** [comment]
@shreys7 @umeshp7 Please review!

**Comment by sastava007** [comment]
@umeshp7 If you are free, then can you please review this one by the time I resolve conflicts in the other PR.

**Comment by sastava007** [comment]
@umeshp7 @codenirvana @abhijitkane Please review!

File: codegens/nodejs-native/lib/request.js [hunk 1]
```diff
@@ -15,19 +15,48 @@ var self;
...
   if (options.followRedirect) {
-    snippet = `var ${nativeModule} = require('follow-redirects').${nativeModule};\n`;
+    if (options.ES6_enabled) {
+      snippet = 'const ';
+    }
+    else {
+      snippet = 'var ';
+    }
+    snippet += `${nativeModule} = require('follow-redirects').${nativeModule};\n`;
+  }
```
Comment (shreys7) [inline codegens/nodejs-native/lib/request.js]: Can you refactor this flow, such that we only test for ES6_enabled only once?
Something like below
```
if (options.ES6_enabled) {
  snippet = 'const ';
}
else {
  snippet = 'var ';
}
if (options.followRedirect) {
 require('follow-redirects).nativeModule;
}
else {
  require('nativeModule');
}
```

File: codegens/nodejs-native/test/unit/snippet.test.js [hunk 2]
```diff
@@ -94,6 +94,43 @@ describe('nodejs-native convert function', function () {
...
+        'url': {
+          'raw': 'https://google.com',
+          'protocol': 'https',
+          'host': [
+            'google',
+            'com'
+          ]
+        }
+      }),
+      nativeModule = (request.url.protocol === 'http' ? 'http' : 'https'),
```
Comment (shreys7) [inline codegens/nodejs-native/test/unit/snippet.test.js]: you don't need to take it from request.url.protocol. directly use `https`

File: codegens/nodejs-native/test/unit/snippet.test.js [hunk 3]
```diff
@@ -94,6 +94,43 @@ describe('nodejs-native convert function', function () {
       expect(snippet).to.include('\'key_containing_whitespaces\': \'  value_containing_whitespaces  \'');
     });
   });
+
+  it('should return snippet with ES6 features when ES6_enabled is set to true', function () {
```
Comment (shreys7) [inline codegens/nodejs-native/test/unit/snippet.test.js]: Can we also test for other changes, i.e. postData, qs, etc.

**Comment by sastava007** [comment]
@shreys7 I have updated everything as you said, please have a look.
🙂

**Pull request merged**